cvat/apps/redis_handler/migration_loader.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (C) 2025 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import importlib
+from pathlib import Path
+from typing import Any, Generator, Mapping
+
+from django.apps import AppConfig, apps
+from django.utils.module_loading import module_has_submodule
+
+from cvat.apps.redis_handler.models import RedisMigration as DBRedisMigration
+from cvat.apps.redis_handler.redis_migrations import BaseMigration as BaseRedisMigration
+
+_MigrationsPerApp = dict[int, list[str]]
+
+
+class LoaderError(Exception):
+    pass
+
+
+class MigrationLoader:
+    REDIS_MIGRATIONS_DIR_NAME = "redis_migrations"
+    REDIS_MIGRATION_CLASS_NAME = "Migration"
+
+    class AppConfigs:
+        def __init__(self):
+            self._app_configs = self._find_app_configs()
+            self._app_configs_mapping: Mapping[str, AppConfig] = self._init_app_configs_mapping()
+
+        def _find_app_configs(self):
+            return [
+                app_config
+                for app_config in apps.get_app_configs()
+                if app_config.name.startswith("cvat")
+                and module_has_submodule(
+                    app_config.module, MigrationLoader.REDIS_MIGRATIONS_DIR_NAME
+                )
+            ]
+
+        def _init_app_configs_mapping(self):
+            return {app_config.label: app_config for app_config in self._app_configs}
+
+        def __getitem__(self, label: str) -> AppConfig:
+            return self._app_configs_mapping[label]
+
+        def __iter__(self) -> Generator[AppConfig, Any, Any]:
+            yield from self._app_configs
+
+    def __init__(self) -> None:
+        self._app_configs = self.AppConfigs()
```

Comment:
Does this need to be a specialized object? It seems like it's only used once, as a map, so it could just be a regular dict.

Reply:
removed